packages/amplication-client/src/ServiceConnections/topics/TopicsList.tsx

```diff
@@ -52,7 +53,7 @@ export default function TopicsList({
                 }
                 onMessagePatternTypeChange={(pattern) => {
                   trackEvent({
-                    eventName: `messagePatternTypeClick-${pattern}`,
+                    eventName: `${AnalyticsEventNames.MessagePatternTypeClick}-${pattern}`,
```

Comment:
@yuval-hazaz we changed the type of the event name to:
```
export interface Event {
  eventName: AnalyticsEventNames | string;
  [key: string]: unknown;
}
```
We want to get rid of the `| string` part, but to do that, we need your insight about transforming this to something like:
```
trackEvent({
  eventName:  AnalyticsEventNames.MessagePatternTypeClick,
  messagePattern: pattern,
})
```
I assume it will affect the data for this event.

Reply:
I dont think there should be an issue. 
the previous type already supported any other properties beside the eventName by using  [key: string]: unknown;
So you only changed eventName from string to enum which should no have any impact on the rest of the properties

Reply:
@yuval-hazaz the only potential issue would be the mis-alignment of historical data since previously tracked events named `messagePatternTypeClick-${pattern}` will become `messagePatternTypeClick` with pattern property

Reply:
I don't think you understood my question. Of course changing the `eventName`'s type will not impact the rest of the properties. The answer is what will happen to the event name "messagePaternClick-send" if we change it to:
```

Reply:
or, to give  more precise example:
```
trackEvent({
  eventName:  "messagePatternTypeClick",
  messagePattern: "send",
})
```

Reply:
I still don't get it :-) let's talk

Reply:
oh... sorry. I just noticed the original code. 
YES. you can definitely change it to the new behavior where the event name is constant for all patterns and the data is send in another prop